pkg/build/api/types.go

```diff
@@ -559,10 +558,35 @@ type BuildConfigSpec struct {
 	// are defined, a new build can only occur as a result of an explicit client build creation.
 	Triggers []BuildTriggerPolicy
 
+	// SchedulePolicy describes how the new build created from this build
+	// configuration will be scheduled for execution.
+	// This is optional, if not specified we default to "parallel".
+	SchedulePolicy *BuildSchedulePolicy
+
 	// BuildSpec is the desired build specification
 	BuildSpec
 }
 
+// BuildSchedulePolicy defines the behaviour of how the new builds are executed
+// from the existing build configuration.
+type BuildSchedulePolicy string
+
+const (
+	// BuildSchedulePolicyParallel schedules new builds immediately after they are
+	// created. Builds will be executed in parallel. This is the default policy.
+	BuildSchedulePolicyParallel BuildSchedulePolicy = "parallel"
+
+	// BuildSchedulePolicySerial schedules new builds to execute in a sequence as
+	// they are created. Every build gets queued up and will execute when the
+	// previous build completes.
+	BuildSchedulePolicySerial BuildSchedulePolicy = "serial"
+
+	// BuildSchedulePolicyRapid schedules new builds to execute in a sequence but
+	// cancelling previous builds if there are any. This allows newly triggered
+	// build to be always executed as "next".
+	BuildSchedulePolicyRapid BuildSchedulePolicy = "rapid"
```

Comment:
Unless we have a really good reason, `SerialMostRecent` or `SerialLatest` or whatever, vs something that doesn't have the `Serial` prefix.